src/view/adminhtml/web/js/filter-attributes.js

```diff
@@ -15,36 +15,47 @@ define([
         },
 
         _initAttributes: function (name) {
-            var category_id = registery.get('emico_attributelanding_page_form.emico_attributelanding_page_form.general.category_id').value();
-            var selectAttribute =  $('select[name="' + name + '"]');
-            var inputAttribute = $('input[name="' + name.replace('[attribute-tmp]', '[attribute]') + '"]');
-            var inputValue = $('input[name="' + name.replace('[attribute-tmp]', '[value]') + '"]');
-            var selectedValue = inputAttribute.val();
-            var facetUrl = url.build('/tweakwise/ajax/facets/category/' + category_id);
-            var foundSelectedValue = false;
+            //bind to event. So it's not triggered by the user, but can be triggered in the code.
+            $('select[name="' + name + '"]').on('initAttributes', function(evt) {
+                var name = evt.target.name;
+                var category_id = registery.get('emico_attributelanding_page_form.emico_attributelanding_page_form.general.category_id').value();
+                var selectAttribute =  $('select[name="' + name + '"]');
+                var inputAttribute = $('input[name="' + name.replace('[attribute-tmp]', '[attribute]') + '"]');
+                var inputValue = $('input[name="' + name.replace('[attribute-tmp]', '[value]') + '"]');
+                var selectedValue = inputAttribute.val();
+                var templateValue = $('select[name="tweakwise_filter_template"]');
+                var facetUrl = url.build('/tweakwise/ajax/facets/category/' + category_id);
+                var foundSelectedValue = false;
 
-            inputAttribute.hide();
-            inputValue.hide();
+                if (templateValue.val() !== '') {
+                    facetUrl += '/filtertemplate/' + templateValue.val();
+                }
+
+                inputAttribute.hide();
+                inputValue.hide();
+
+                $.getJSON(facetUrl, function( data ) {
+                    selectAttribute.empty();
+                    data.data.forEach(value => {
```

Comment:
On my side, `data.data` seems to be an object. Which you cannot loop through using .forEach()
![data data_error](https://github.com/EmicoEcommerce/Magento2AttributeLandingTweakwise/assets/4329209/3a6271da-f2a5-4e4f-86da-d9c83e06328a)

Reply:
Issue is resolved in the other pull requests. https://github.com/EmicoEcommerce/Magento2Tweakwise/pull/116/commits/e8d6a01b0a4e46d27677a1b57d32daebf4b6ba11